apps/prairielearn/src/components/PageLayout.tsx

```diff
@@ -102,6 +108,44 @@ function SyncErrorsAndWarningsComponent({
   }
 }
 
+function UnpublishedBannerComponent({
+  navContext,
+  resLocals,
+}: {
+  navContext: NavContext;
+  resLocals: UntypedResLocals;
+}) {
+  if (navContext.type !== 'instructor') return null;
+  if (navContext.page === 'instance_admin' && navContext.subPage === 'publishing') return null;
```

Comment:
These conditionals don't take into account the fact that course admin pages are accessible from course instance routes:

- Click into a course instance
- Click "course settings" on the sidebar

The alert is shown. While you're "technically" looking at a course instance route, the page itself is associated with the course, so I feel it's incorrect to show it on course pages.

Reply:
With the updated check:

```ts
  if (!['instance_admin', 'assessment'].includes(navContext.page)) {
    return null;
  }
```

I have tested this again and I can't see the banner.

<img width="2422" height="1172" alt="CleanShot 2025-12-09 at 18 51 03@2x" src="https://github.com/user-attachments/assets/18e5a9a9-86a9-4d32-8df4-232d61cb4552" />